components/src/components/shared/checkout-dialog/Plan.tsx

```diff
@@ -133,109 +136,154 @@ const PlanButtonGroup = ({
-                  type="button"
-                  disabled={isLoading || !isValidPlan}
-                  onClick={() => {
-                    onSelect({ plan, shouldTrial: false });
-                  }}
-                  $size="sm"
-                  $color="primary"
-                  $variant={isTrialing ? "filled" : "text"}
-                  $fullWidth
-                >
-                  {!isValidPlan ? (
-                    <Box $textAlign="center">
-                      <Text as={Box} $align="center">
-                        {t("Over plan limit")}
-                      </Text>
-                    </Box>
-                  ) : (
-                    t("Choose plan")
-                  )}
-                </Button>
+              </Button>
 
-                {!plan.valid && (
-                  <UsageViolationText violations={plan.usageViolations} />
-                )}
-              </Flex>
-            )}
-          </>
-        )}
-      </Flex>
-    );
-  }
+              {!plan.valid && (
+                <UsageViolationText violations={plan.usageViolations} />
+              )}
+            </Flex>
+          )}
+        </>
+      )}
+    </Flex>
+  );
+};
+
+export const ChoosePlan = ({
+  plan,
+  isLoading,
+  isSelected,
+  onSelect,
+}: SharedPlanProps) => {
+  const { t } = useTranslation();
+
+  const { data } = useEmbed();
+
+  const isDowngradeNotPermitted =
+    plan.invalidReason === CompanyPlanInvalidReason.DowngradeNotPermitted;
+  const isDisabled =
+    (isLoading || (!plan.valid && !isDowngradeNotPermitted)) && !plan.custom;
 
   return isSelected ? (
-    <Selected isCurrent={isCurrentPlan} />
+    <Selected isCurrent={plan.current} />
   ) : (
     <Flex $flexDirection="column" $gap="0.5rem">
       <Button
         type="button"
-        disabled={(isLoading || !isValidPlan) && !plan.custom}
+        disabled={isDisabled}
         {...(plan.custom
           ? {
               as: "a",
               href: plan.customPlanConfig?.ctaWebSite ?? "#",
               target: "_blank",
               rel: "noreferrer",
             }
-          : {
-              onClick: () => {
-                onSelect({ plan });
-              },
-            })}
+          : isDowngradeNotPermitted
+            ? {
+                as: "a",
+                href: data?.preventSelfServiceDowngradeUrl ?? "#",
+                target: "_blank",
+                rel: "noreferrer",
+              }
+            : {
+                onClick: () => {
+                  onSelect({ plan });
+                },
+              })}
         $size="sm"
         $color="primary"
         $variant="filled"
         $fullWidth
       >
-        {plan.custom ? (
-          (plan.customPlanConfig?.ctaText ?? t("Talk to support"))
-        ) : !isValidPlan ? (
-          <Text as={Box} $align="center">
-            {t("Over plan limit")}
-          </Text>
-        ) : (
-          t("Choose plan")
-        )}
+        {plan.custom
```

Comment:
This ternary logic seems to be inconsistent with other similar cases - is it correct?